contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -804,8 +814,9 @@ void USDScene::writeAttribute( const SceneInterface::Name &name, const Object *a
 		if( const Data *data = runTimeCast<const Data>( attribute ) )
 		{
 			pxr::UsdAttribute attribute = m_location->prim.CreateAttribute(
-				pxr::TfToken( name.string() ),
-				DataAlgo::valueTypeName( data )
+				pxr::TfToken( SceneCacheDataAlgo::toPrimVarName( name.string() ) ),
+				DataAlgo::valueTypeName( data ),
+				/*custom=*/true
```

Comment:
I'm not sure if I understand the comment correctly but it sounds like a misunderstanding ( related to your other comment regarding the naming of `SceneCacheDataAlgo::toPrimVarName` ), this is not creating a new USD primvar but rather creating a new custom USD Attribute ( that might comes from a Cortex Attribute or  Cortex Constant PrimitiveVariable ) so it needs renaming.

Further more, the previous code use to create Attribute with `custom=false` which was hiding the bug we had in `isCustomAttribute` but something like Houdini would create attribute as `custom` as it should.
Since we hit the `else if` clause after the non custom attribute, then it should definitely be a custom attribute.

I hope that makes sense but let me know if I misunderstood something.

Reply:
We've called `toPrimVarName()` to get the name of the attribute to write, which means we may be writing an attribute called `primvars:user:foo`. Writing that via `CreateAttribute()` isn't exactly equivalent to writing it via PrimvarsAPI. From memory (I looked at this yesterday), the former created a custom attribute, and the latter created a non-custom attribute with an additional interpolation field.

This may not matter if the attribute we write is sufficient to fool the PrimvarsAPI on reading (it seems to be), but it does result in a slightly non-standard looking USD file, and may make us vulnerably to changes in the PrimvarsAPI in future.

Reply:
addressed here 86c6efaf3cf412b8aed70f0d27d5beb1c578c835 with a follow up for non `user/render` prefixed attribute here 5533caea737c3d27156b4d119aa9ab8b04e02d4a